kube/src/api/subresource.rs

```diff
@@ -208,3 +211,177 @@ where
         Ok(self.client.request_text_stream(req).await?)
     }
 }
+
+// ----------------------------------------------------------------------------
+// Attach subresource
+// ----------------------------------------------------------------------------
+/// Params for attaching
+///
+/// Note that the server rejects when none of `stdin`, `stdout`, `stderr` are attached.
+#[cfg(feature = "ws")]
+pub struct AttachParams {
+    /// The container in which to execute the command.
+    /// Defaults to only container if there is only one container in the pod.
+    pub container: Option<String>,
+    /// If `true`, the contents will be redirected to the standard input stream of the pod. Defaults to `false`.
+    pub stdin: bool,
+    /// If `true`, the standard out stream of the pod will be redirected to it. Defaults to `true`.
+    pub stdout: bool,
+    /// If `true`, the standard error stream of the pod will be redirected to it. Defaults to `true`.
+    pub stderr: bool,
```

Comment:
`stdin`/`stdout`/`stderr` are now booleans.

Reply:
makes sense, guess they just go into the query params as is.

Reply:
Though, afaikt the query parameters only technically enable the stdout and stderr channels, right?
If that's true, the comments should probably reflect that.